### ISSUE TITLE
Group issues after update 1.6 to 2.0.7

### ISSUE THREAD

**Issue by DanielBoettner** [passbolt/passbolt_api#261]
## Group issues after update 1.6 to 2.0.7
* Passbolt Version: 2.0.7
* Platform and Target:
-- Operating system: Debian Linux (docker container)
-- PHP: 7.2.5
-- Database server: MySQL 5.7.18

### What you did

Updated the Passbolt installation from 1.6.x to 2.0.7.
Ran migration commands via console.

Login to Passbolt, go to **users** and edit a group (preferably one that was existing before an update).
When we try adding a new user to the group it fails.

### What happened
In a specific case the group has 4 users already. When trying to add a new user the UI communicates that `decrypting 4/5` and a progress of 8%. 

` Uncaught (in promise) TypeError: Cannot read property 'Secret' of undefined ` when trying to add a user to a group, in which I was not previously a group_manager.

### What you expected to happen
Passbolt adds the user to the group.

### General comment

I',m still not sure if it is the correct place.

After I switched our previous (docker container) with a newer version. 1.6.x to 2.0.7.
It works in general but has bugs/misbehavior like Issue #249.

As commented, the group functionality seems to have issues.

In short, I have an error when I try to add a user to a group that was existing before my last update.
I thought about creating a new group and changing the owner of the passwords to the new group.
But I don't know if this would really solve the problems.

And a bad request when trying to add a user to a group I always was managing.

`vendors.min.js:82340 PUT https://mydomain.tld:4343/groups/5a1d50a9-24c0-4525-878a-0096ac11000c.json?api-version=v1 400 (Bad Request)`

**Comment by stripthis** [comment]
Can you confirm you get nothing when running
```
./bin/cake passbolt cleanup
```

**Comment by DanielBoettner** [comment]
# Cleanup Call 

 `su -s /bin/bash -c "/var/www/passbolt/bin/cake passbolt cleanup" www-data`

# Result
```
 Cleanup shell (delete mode)
---------------------------------------------------------------
No issue found, data looks squeaky clean!
```

# Healthcheck
## Call

` su -s /bin/bash -c "/var/www/passbolt/bin/cake passbolt healthcheck" www-data`

## Result

3 Fails related to SSL and healthcheck/status ,which is likely caused by the way the page is accessed (via a port). 

* Healthcheck url not reachable
* SSL peer certificate does not validate
* Hostname does not match

**Comment by stripthis** [comment]
To me it looks like you have one permission with no associated secret (e.g. 5 permissions and 4 secrets). I'm not sure how you got into that state as it should not be possible (unless you manually edited the db at some point). 

Send me an email at contact@passbolt.com if you want we can get on a call and check that.

**Comment by DanielBoettner** [comment]
I indeed had to edit the DB manually.
I had to add myself to a group with elevated permissions.

I will try to move the existing passwords to a new group.
And test if it works.

Maybe my update process was just messy at some point.

I'll keep you updated.

**Comment by stripthis** [comment]
The way to do it would be to elevate the permission of somebody that is already in the group so you don't have that problem. I'm closing this because this is an integrity issue due to manual editing of the Database, which is not something supported. Good luck!
